common/src/main/java/org/vivecraft/client_vr/provider/openxr/MCOpenXR.java

```diff
@@ -0,0 +1,1494 @@
+            applicationInfo.applicationVersion(1);
+
+            //Create instance info
+            XrInstanceCreateInfo createInfo = XrInstanceCreateInfo.calloc(stack);
+            createInfo.type(XR10.XR_TYPE_INSTANCE_CREATE_INFO);
+            createInfo.next(NULL);
+            createInfo.createFlags(0);
+            createInfo.applicationInfo(applicationInfo);
+            createInfo.enabledApiLayerNames(null);
+            createInfo.enabledExtensionNames(extensions.flip());
+
+            //Create XR instance
+            PointerBuffer instancePtr = stack.callocPointer(1);
+            int xrResult = XR10.xrCreateInstance(createInfo, instancePtr);
+            if (xrResult == XR10.XR_ERROR_RUNTIME_FAILURE) {
+                throw new RuntimeException("Failed to create xrInstance, are you sure your headset is plugged in?");
+            } else if (xrResult == XR10.XR_ERROR_INSTANCE_LOST) {
+                throw new RuntimeException("Failed to create xrInstance due to runtime updating");
+            } else if (xrResult < 0) {
+                throw new RuntimeException("XR method returned " + xrResult);
+            }
+            instance = new XrInstance(instancePtr.get(0), createInfo);
+
+            this.poseMatrices = new Matrix4f[64];
+
+            for (int i = 0; i < this.poseMatrices.length; ++i) {
+                this.poseMatrices[i] = new Matrix4f();
+            }
+
+            this.initSuccess = true;
+        }
+    }
+
+    public static MCOpenXR get() {
+        return ome;
+    }
+
+    private void initializeOpenXRSession() {
+        try (MemoryStack stack = MemoryStack.stackPush()) {
+            //Create system
+            XrSystemGetInfo system = XrSystemGetInfo.calloc(stack);
+            system.type(XR10.XR_TYPE_SYSTEM_GET_INFO);
+            system.next(NULL);
+            system.formFactor(XR10.XR_FORM_FACTOR_HEAD_MOUNTED_DISPLAY);
+
+            LongBuffer longBuffer = stack.callocLong(1);
+            int error = XR10.xrGetSystem(instance, system, longBuffer);
+            logError(error, "xrGetSystem",  "");
+            this.systemID = longBuffer.get(0);
+
+            if (systemID == 0) {
+                throw new RuntimeException("No compatible headset detected");
+            }
+
+            //Bind graphics
+            Struct graphics = this.getGraphicsAPI(stack);
+
+            //Create session
+            XrSessionCreateInfo info = XrSessionCreateInfo.calloc(stack);
+            info.type(XR10.XR_TYPE_SESSION_CREATE_INFO);
+            info.next(graphics.address());
+            info.createFlags(0);
+            info.systemId(systemID);
+
+            PointerBuffer sessionPtr = stack.callocPointer(1);
+            error = XR10.xrCreateSession(instance, info, sessionPtr);
+            logError(error, "xrCreateSession",  "");
+
+            session = new XrSession(sessionPtr.get(0), instance);
+
+            XrSessionBeginInfo sessionBeginInfo = XrSessionBeginInfo.calloc(stack);
```

Comment:
Results in XR_ERROR_SESSION_NOT_READY as you did not wait for the runtime signal.